handlers/stallRequestHandler.js

```diff
@@ -0,0 +1,86 @@
+import StallRequest from "../models/StallRequest";
+import { StatusCodes } from "http-status-codes";
+
+const stallRequestHandler = async (req, res) => {
+  if (req.method === "GET") {
+    const { vendorId } = req.query;
+
+    const stalls = await StallRequest.find({
+      vendorId: vendorId,
+    }).exec();
+
+    if (!stalls) {
+      res
+        .status(StatusCodes.NOT_FOUND)
+        .json({ message: "Stall request not found" });
+      return;
+    }
+
+    res.status(StatusCodes.OK).json({ stalls });
+  }
+
+  if (req.method === "POST") {
+    const { vendorId, eventId, stallId } = req.body;
+
+    const request = new StallRequest({
+      vendorId,
+      eventId,
+      stallId,
+    });
+
+    request
+      .save()
+      .then(() => {
+        res
+          .status(StatusCodes.CREATED)
+          .json({ message: "Stall request inserted successfully." });
+      })
+      .catch((err) => {
+        res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({
+          message: "Unable to save stall request",
+          error: err,
+        });
+      });
+  }
+
+  if (req.method === "PATCH") {
+    const { stallRequestId } = req.query;
+
+    StallRequest.findByIdAndUpdate(
```

Comment:
what does this one do?

Reply:
We will be using it to update the status of a request.